R/proxy.R

```diff
@@ -0,0 +1,27 @@
+#' tidyselect methods for custom types
+#'
+#' * `tidyselect_data_proxy()` should return a 0-row data frame.
```

Comment:
Ooops, you're right.

We could make this a hierarchy, but we'd need to use some heuristic to figure out if a predicate was a type function or something else. So while it would be a nice improvement to the interface, I don't think it's worth the effort.